File: src/main/kotlin/com/annotation/log/logutil/util/LoggerFactory.java
package com.annotation.log.logutil.util;

import org.slf4j.Logger;

/**
 * @ClassName LoggerFactory
 * @Description TODO
 * @Author pei.zhang
 * @Date 2018/10/23
 * @Version 1.0
 */
public class LoggerFactory {

    private static Logger logger;
    public static com.annotation.log.logutil.util.Logger getLogger(Class<?> clazz) {
        logger = org.slf4j.LoggerFactory.getLogger(clazz);
        com.annotation.log.logutil.util.Logger log = new com.annotation.log.logutil.util.Logger(logger);
        return log;
    }
}

File: src/main/kotlin/com/annotation/log/logutil/util/LoggerFactory.kt
package com.annotation.log.logutil.util


/**
 * @ClassName LoggerFactory
 * @Description 代理LoggerFactory类
 * @Author pei.zhang
 * @Date 2018/10/23
 * @Modified 2019-1-5
 * @Version 1.4
 */
@Deprecated("please use MDC binging traceId")
object LoggerFactory {

    private var logger: org.slf4j.Logger? = null
    @JvmStatic fun getLogger(clazz: Class<*>): Logger {
        logger = org.slf4j.LoggerFactory.getLogger(clazz)
        val log : Logger = Logger(logger!!)
        return log
    }
}
